terraform/lib/dependabot/terraform/file_updater.rb

```diff
@@ -81,6 +91,42 @@ def update_registry_declaration(new_req, old_req, updated_content)
         end
       end
 
+      def update_lockfile_declaration(dependency)
```

Comment:
We're calling this twice, redoing all that work each time, we can solve this in two ways, the first one is to memoize the result of this method, in that case we need to account for the fact that we pass it a `dependency` arg, so:

```ruby
def update_lockfile_declaration(dependency)
  @updated_lockfile_declarations[dependency] ||=
    begin
      # .. the current implementation
    end
end
```

Or we can make sure that we only call it once, by assigning the results of the method in `#updated_dependency_files`, and passing it as an argument to `lockfile_changed?`

Reply:
Implemented in https://github.com/dependabot/dependabot-core/pull/3766/commits/41930eff1c4a1b2ca7e37f3ad57591a30d5f5d38.